pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -559,6 +571,60 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         applyNCMultiQubitOp(arr, num_qubits, matrix, {}, {}, wires, inverse);
     }
 
+    template <class PrecisionT>
+    static void
+    applyPauliRot(std::complex<PrecisionT> *arr, std::size_t num_qubits,
+                  const std::vector<std::size_t> &wires, const bool inverse,
+                  PrecisionT angle, const std::string &word) {
+        using ComplexT = std::complex<PrecisionT>;
+        constexpr auto IMAG = Pennylane::Util::IMAG<PrecisionT>();
+        PL_ABORT_IF_NOT(wires.size() == word.size(),
+                        "wires and word have incompatible dimensions.")
+        const PrecisionT c = std::cos(angle / 2);
+        const ComplexT s = ((inverse) ? IMAG : -IMAG) * std::sin(angle / 2);
+        const std::array<ComplexT, 4> sines{s, IMAG * s, -s, -IMAG * s};
+
+        auto get_mask =
+            [num_qubits,
+             &wires]([[maybe_unused]] const std::function<bool(const int)>
+                         &condition) {
+                std::size_t mask{0U};
+                for (std::size_t iw = 0; iw < wires.size(); iw++) {
+                    const auto bit = static_cast<std::size_t>(condition(iw));
+                    mask |= bit << (num_qubits - 1 - wires[iw]);
+                }
+                return mask;
+            };
+        const std::size_t mask_xy =
+            get_mask([&word](const int a) { return word[a] != 'Z'; });
+        const std::size_t mask_y =
+            get_mask([&word](const int a) { return word[a] == 'Y'; });
+        const std::size_t mask_z =
+            get_mask([&word](const int a) { return word[a] == 'Z'; });
+        const auto count_mask_y = std::popcount(mask_y);
+        const std::size_t sv_length = exp2(num_qubits);
+
+        PL_LOOP_PARALLEL(1)
+        for (std::size_t i0 = 0; i0 < sv_length; i0++) {
+            const std::size_t i1 = i0 ^ mask_xy;
+            if (i0 > i1) {
+                continue;
+            }
+            const auto count_y = std::popcount(i0 & mask_y) * 2;
+            const auto count_z = std::popcount(i0 & mask_z) * 2;
+            const auto sign_i0 = count_z + count_mask_y * 3 - count_y;
+            if (mask_xy) [[likely]] {
+                const auto sign_i1 = count_z + count_mask_y + count_y;
+                const ComplexT v0 = arr[i0];
+                const ComplexT v1 = arr[i1];
+                arr[i0] = c * v0 + sines[sign_i0 % 4] * v1;
+                arr[i1] = c * v1 + sines[sign_i1 % 4] * v0;
+            } else {
+                arr[i0] *= c + sines[sign_i0 % 4];
+            }
```

Comment:
```suggestion
            if (mask_xy) [[likely]] {
                const auto sign_i1 = count_z + count_mask_y + count_y;
                const ComplexT v0 = arr[i0];
                const ComplexT v1 = arr[i1];
                arr[i0] = c * v0 + sines[sign_i0 % 4] * v1;
                arr[i1] = c * v1 + sines[sign_i1 % 4] * v0;
            } else [[unlikely]] {
                arr[i0] *= c + sines[sign_i0 % 4];
            }
```
Also it is necessary to specify the `unlikely` :sweat_smile:

Reply:
Thanks for bringing this up. This is our first use of it in the code base I reckon.